web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,232 @@
+            this.$appLinks.on('mouseenter', $.proxy(this.selectAppLinkEvent, this));
+            this.$el.one('drawer.closed', $.proxy(this.onDrawerClose, this));
+            core.bus.trigger('drawer.opened');
+        },
+        
+        // It provides a proxy method allowing for app link select w/ events
+        selectAppLinkEvent: function(event) {
+            this.selectAppLink($(event.currentTarget));
+        },
+        
+        // It selects an app link visibly
+        selectAppLink: function($appLink) {
+            if ($appLink) {
+                this.$selectedAppLink = $appLink;
+                $appLink.focus();
+            }
+        },
+        
+        /* It searches for app links by name and returns the first match
+         * @param query str to search
+         * @return jQuery obj
+         */
+        searchAppLinks: function(query) {
+            return this.$appLinks.filter(function() {
+                return $(this).data('menuName').toUpperCase().startsWith(query);
+            }).first();
+        },
+        
+        /* Find the link adjacent to $appLink in provided direction
+         * @param $appLink jQuery obj of App icon link
+         * @param direction Array for direction to look [int:x, int:y]
+         * @return jQuery obj for adjacent applink
+         */
+        findAdjacentAppLink: function($appLink, direction) {
+            
+            var obj = [],
+                $objs = this.$appLinks;
+            
+            switch(direction){
+                case this.LEFT:
+                    obj = $objs[$objs.index($appLink) - 1];
+                    if (!obj) {
+                        obj = $objs[$objs.length - 1];
+                    }
+                    break;
+                case this.RIGHT:
+                    obj = $objs[$objs.index($appLink) + 1];
+                    if (!obj) {
+                        obj = $objs[0];
+                    }
+                    break;
+                case this.UP:
+                    $objs = this.getRowObjs($appLink, this.$appLinks);
+                    obj = $objs[$objs.index($appLink) - 1];
+                    if (!obj) {
+                        obj = $objs[$objs.length - 1];
+                    }
+                    break;
+                case this.DOWN:
+                    $objs = this.getRowObjs($appLink, this.$appLinks);
+                    obj = $objs[$objs.index($appLink) + 1];
+                    if (!obj) {
+                        obj = $objs[0];
+                    }
+                    break;
+            }
+            
+            return $(obj);
+            
+        },
+        
+        /* Return els in the same row
+         * @param @obj jQuery object to get row for
+         * @param $grid jQuery objects representing grid
+         * @return $objs jQuery objects of row
+         */
+        getRowObjs: function($obj, $grid) {
+            // Filter by object which middle lies within left/right bounds
```

Comment:
I think this would be more accurate if you fix a little wrong alignment glitch that there's right now on runbot:

![captura de pantalla de 2016-08-25 10-24-53](https://cloud.githubusercontent.com/assets/973709/17961957/c743ee82-6aae-11e6-80aa-8fcc584de672.png)